modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/AsyncContinuation.kt

```diff
@@ -19,15 +17,12 @@ typealias Disposable = () -> Unit
 
 @RestrictsSuspension
 @Suppress("DELEGATED_MEMBER_HIDES_SUPERTYPE_OVERRIDE")
-open class MonadDeferCancellableContinuation<F, A>(val SC: MonadDefer<F>, override val context: CoroutineContext = EmptyCoroutineContext) :
-  MonadErrorContinuation<F, A>(SC), MonadDefer<F> by SC, BindSyntax<F> {
+open class AsyncContinuation<F, A>(val SC: Async<F>, override val context: CoroutineContext = EmptyCoroutineContext) :
```

Comment:
For another diff we should interface this, instead of passing the full class have a new interface that composes `Async<F> by SC, BindSyntax<F>` or whatever.